packages/teleport/src/services/recordings/makeRecording.ts

```diff
@@ -1,7 +1,50 @@
 import moment from 'moment';
-import { Recording } from './types';
+import { Recording, RecordingType, RecordingDescription } from './types';
+import { eventCodes } from 'teleport/services/audit';
 
-export default function makeRecording({
+// Takes in json objects built by SessionEnd and WindowsDesktopSessionEnd as defined in teleport/api/types/events/events.proto.
+export function makeRecording(event: any): Recording {
+  if (event.code === eventCodes.DESKTOP_SESSION_ENDED) {
+    return makeDesktopRecording(event);
+  } else {
+    return makeSshRecording(event);
+  }
+}
+
+function makeDesktopRecording({
+  time,
+  session_start,
+  session_stop,
+  user,
+  sid,
+  desktop_name,
+  session_recording,
+}) {
+  const { duration, durationText } = formatDuration(
+    session_start,
+    session_stop
+  );
+
+  let description: RecordingDescription = 'play';
```

Comment:
why typing a text?

Reply:
Oh, i see why. This is because the same description text is used for 2 event types. In this case i think it's OK to keep it.

Reply:
My editor was complaining at me if I didn't do it like this 🤷